l3kernel/l3int.dtx

```diff
@@ -307,6 +307,35 @@
 %   \cs{int_eval:n}).
 % \end{function}
 %
+% \begin{function}[added = 2024-11-26]
+%   {
+%     \int_set_regex_count:Nnn, \int_set_regex_count:cnn,
+%     \int_set_regex_count:NNn, \int_set_regex_count:cNn,
+%     \int_gset_regex_count:Nnn, \int_gset_regex_count:cnn,
+%     \int_gset_regex_count:NNn, \int_gset_regex_count:cNn,
+%   }
+%   \begin{syntax}
+%     \cs{int_set_regex_count:Nnn} \meta{int var} \Arg{regex} \Arg{token list}
+%     \cs{int_set_regex_count:NNn} \meta{int var} \Arg{regex~var} \Arg{token list}
+%   \end{syntax}
+%   Sets \meta{int var} equal to the number of times
+%   \meta{regular expression} appears in \meta{token list}.
+%   The search starts by finding the left-most longest match,
+%   respecting greedy and lazy (non-greedy) operators. Then the search
+%   starts again from the character following the last character
+%   of the previous match, until reaching the end of the token list.
+%   Infinite loops are prevented in the case where the regular expression
+%   can match an empty token list: then we count one match between each
+%   pair of characters.
+%   For instance,
+%   \begin{verbatim}
+%     \int_set_regex_count:Nnn \l_foo_int { (b+|c) } { abbababcbb }
+%   \end{verbatim}
+%   results in \cs[no-index]{l_foo_int} taking the value $5$.
+%   See \pkg{l3regex} chapter for more details of the \meta{regex}
+%   format.
```

Comment:
Another general idea (sorry, didn't come up in the first round): as they are thin wrappers, do we want to clarify their relationship with `\regex_...` functions? Like

"Theses are alternative names for \cs{regex_count:nnN} and friends, with arguments re-ordered for \meta{int~var} setting."